src/stanza.c

```diff
@@ -431,7 +431,7 @@ int xmpp_stanza_to_text(xmpp_stanza_t *stanza,
     ret = _render_stanza_recursive(stanza, buffer, length);
     if (ret < 0) return ret;
 
-    if (ret > length - 1) {
+    if ((unsigned)ret > length - 1) {
```

Comment:
Please, replace unsigned with size_t here and below. Libstrophe can be built for different platforms and C standard doesn't require `size_t` to be `unsigned int` (AFAIR).